### PR TITLE
Fix adjustMergeCells not modifies cell rect

### PR DESCRIPTION
**Pull request by Dokiys**
Fix DuplicateRowTo unvalied thad duplicates later row to forehead

# PR Details

Fix adjustMergeCells not modifies cell rect and DuplicateRowTo unvalied thad duplicates later row to forehead

## Description

`MergeCell` after `DupliacteRowTo` will get unexpected result. Such as below cases: 
```go
         // Test merge cell after duplicate null row
	{
		f := excelize.NewFile()
		sheet := "Sheet1"
		assert.NoError(t, f.MergeCell(sheet,"A4","A5"))

		assert.NoError(t, f.DuplicateRowTo(sheet, 2, 3))
		assert.NoError(t, f.DuplicateRowTo(sheet, 2, 4))
		assert.NoError(t, f.MergeCell(sheet,"A2","A3"))
		assert.NoError(t, f.MergeCell(sheet,"A3","A4"))

		mergeCells, err := f.GetMergeCells(sheet)
		assert.NoError(t, err)
		assert.ElementsMatch(t, mergeCells, []excelize.MergeCell{{"A6:A7",""},{"A2:A4",""}})
	}

	// Test merge cell after duplicate null col
	{

		f := excelize.NewFile()
		sheet := "Sheet1"

		assert.NoError(t, f.MergeCell(sheet,"D1","E1"))

		assert.NoError(t, f.InsertCol(sheet, "C"))
		assert.NoError(t, f.InsertCol(sheet, "D"))
		assert.NoError(t, f.MergeCell(sheet,"B1","C1"))
		assert.NoError(t, f.MergeCell(sheet,"C1","D1"))

		mergeCells, err := f.GetMergeCells(sheet)
		assert.NoError(t, err)
		assert.ElementsMatch(t, mergeCells, []excelize.MergeCell{{"B1:D1",""},{"F1:G1",""}})
	}
```
Related unit test have been added in test file.

## How Has This Been Tested

Run unit testing.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x ] I have read the **CONTRIBUTING** document.
- [ x ] I have added tests to cover my changes.
- [ x ] All new and existing tests passed.

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -661,7 +661,8 @@ func (f *File) DuplicateRowTo(sheet string, row, row2 int) error {
 	if err != nil {
 		return err
 	}
-	if row > len(ws.SheetData.Row) || row2 < 1 || row == row2 {
+
+	if row2 < 1 || row == row2 {
 		return nil
```
Comment (xuri) [inline rows.go:666]: Could you add unit test for this exception case?

File: rows.go [hunk 2]
```diff
@@ -675,14 +676,15 @@ func (f *File) DuplicateRowTo(sheet string, row, row2 int) error {
 			break
 		}
 	}
-	if !ok {
-		return nil
-	}
 
 	if err := f.adjustHelper(sheet, rows, row2, 1); err != nil {
 		return err
```
Comment (xuri) [inline rows.go:681]: The unit test doesn't cover this case currently, please add the test for this.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your PR, I've left some comments. Please follows the code style of this project, and lint the code with `go fmt -s`.

**Review by xuri — APPROVED** [review]

**Comment by xuri** [comment]
I made some changes based on your branch.

**Comment by Dokiys** [comment]
> I made some changes based on your branch.

OK. Thanks.

**Pull request merged**